F/ff curl temp

ebranlard opened the pull request:
**Complete this sentence**
THIS PULL REQUEST __ [IS/IS NOT] __ READY TO MERGE

**Feature or improvement description**
<A clear and concise description of the new code.>

**Related issue, if one exists**
<Link to a related GitHub Issue.>

**Impacted areas of the software**
<List any modules or other areas which should be impacted by this pull request. This helps to determine the verification tests.>

**Additional supporting information**
<Add any other context about the problem here.>

**Test results, if applicable**
<Add the results from unit tests and regression tests here along with justification for any failing test cases.>

ebranlard committed:
Curl: dummy cartesian implementation with VTK outputs (gradient terms, curl and swirl todo)

tonyinme committed:
Curl: First implementation of curled wake model

ebranlard committed:
Curl: implemented gradients

ebranlard committed:
Curl: temporarily making wakedynamics depend on aerodyn for VTK

ebranlard committed:
Merge branch 'f/ff-curl' into f/ff-curl-temp

tonyinme committed:
Plane output location changed

ebranlard committed:
Curl: using Ct avg

ebranlard committed:
Merge branch 'f/ff-curl-temp' of https://github.com/ebranlard/openfast into f/ff-curl-temp

tonyinme committed:
Curl: First working version of the curled wake model

tonyinme committed:
Curl: Added swirl - bug in cq computation needs to be fixed

tonyinme committed:
Curl: Fixed Cq calculation

tonyinme committed:
Curl: filtered angle, curl plane vector normal to x, fixed eddy visc

jjonkman commented on glue-codes/fast-farm/src/FASTWrapper.f90:
```diff
@@ -646,12 +655,12 @@ SUBROUTINE FWrap_CalcOutput(p, u, y, m, ErrStat, ErrMsg)
          num = 0.0_ReKi
          do k=1,size(m%ADRotorDisk) ! loop on blades force contribution
-            num = num + dot_product(y%xHat_Disk, m%ADRotorDisk(k)%Moment(:,j) ) 
+            num = num - p%r(j)*dot_product(m%yHat_blade(:,k), m%ADRotorDisk(k)%Force(:,j)  ) + dot_product(y%xHat_Disk, m%ADRotorDisk(k)%Moment(:,j) ) 
```
There is no need to calculate and store m%yHat_blade(:,k) based on a cross product.  This data (yhat^Disk_k,R) is already stored in m%ADRotorDisk(k)%RefOrientation(2,:,1), computed early in the routine.  So, this line can be replaced with:

num = num - p%r(j)*dot_product(m%ADRotorDisk(k)%RefOrientation(2,:,1), m%ADRotorDisk(k)%Force(:,j) ) + dot_product(y%xHat_Disk, m%ADRotorDisk(k)%Moment(:,j) )

jjonkman commented on modules/wakedynamics/src/WakeDynamics.f90:
```diff
@@ -917,19 +1028,293 @@ subroutine updateVelocityPolar()
+
+
+!> Weighted average of two angles
+subroutine filter_angles2(psi_filt, chi_filt, psi, chi, alpha, alpha_bar)
```
This SUBROUTINE looks like exactly what I had laid out.  How do the results look?  Does theta(2) show up as nonzero?

jjonkman commented on modules/wakedynamics/src/WakeDynamics.f90:
```diff
@@ -917,19 +1028,293 @@ subroutine updateVelocityPolar()
+   real(ReKi), intent(in) :: chi !< skew angle
+   real(ReKi), intent(in) :: alpha     !< filter weight
+   real(ReKi), intent(in) :: alpha_bar !< 1-alpha
+   real(ReKi) :: t_filt !< output
```
Variables t_filt and x,y dont appear to be used and can be removed.

jjonkman reviewed (commented):
Thanks, @ebranlard!  I reviewed the updated calculation of Cq and the new skew angle filter.  Please find my comments attached.

tonyinme committed:
Curl: using orientation instead of yhat blade for Cq calculation

tonyinme committed:
Curl: option to keep cross flow in or our of Vdist_low in AWAE

tonyinme committed:
Curl: using absolute value in eddy viscosity

tonyinme committed:
Curl: using Mod_Deficit=1 (backward compatibility, cross flow in Vlow_dist

tonyinme committed:
Curl: latest version of algorithm inncluding 4 main changes
- dx using non filtered velocity
- curled equations in disk normal coordinates
- simple filter angle functions
- initial condition wake filtering
- using Mod_ddeficit=2, which removes the cross flow components from the AWAE wake convection velocity
